src/main/java/edu/hm/hafner/analysis/parser/MentorParser.java

```diff
@@ -0,0 +1,143 @@
+    private static final Pattern VLOG_PATTERN = Pattern.compile(VLOG_REGEX);
+    
+    /**
+     * The first capture group matches the timestamp for the message on the previous line.
+     * The iteration is ignored.
+     * \w* matches a word such as "module" or "protected".
+     * The next capture group captures the path of the module.
+     * The next capture groups capture the File name and Line number.
+     */
+    private static final String TIME_FILE_REGEX = "# {4}Time: (\\d* \\ws)(?: {2}Iteration: \\d+)? {2}\\w*: (.\\S*)(?: File: (\\S+)(?: Line: (\\d+))?)?.*";
+    private static final Pattern TIME_FILE_PATTERN = Pattern.compile(TIME_FILE_REGEX);
+
+    
+    /**
+     * Construct a parser for MentorGraphics Modelsim/Questa logs.
+     */
+    public MentorParser() {
+        super(MSG_REGEX);
+    }
+
+    @Override
+    protected Optional<Issue> createIssue(final Matcher matcher, final LookaheadStream lookahead,
+                                          final IssueBuilder builder) {
+
+        builder.setModuleName(null);
+        builder.setFileName(null);
+        builder.setLineStart(null);
+        builder.setCategory(null);
+
+        String priority = matcher.group(1);
+        String message = matcher.group(2);
+
+        StringBuilder description = new StringBuilder();
+        String timeLine = "";
+
+        if (message.contains("while parsing file")) {
+            while (!lookahead.peekNext().startsWith("# ** at ")) {
+                lookahead.next();
+            }
+            Matcher at = VLOG_PATTERN.matcher(lookahead.next().substring(5));
+            if (at.matches()) {
+                builder.setFileName(at.group(1));
+                builder.setLineStart(at.group(2));
+                builder.setCategory(at.group(3));
+                message = at.group(4);
+            }
+
+        }
+        else {
+            while (lookahead.hasNext() && !lookahead.peekNext().contains("# **")) {
+                timeLine = lookahead.next();
+                if (timeLine.startsWith("#    Time:")) {
+                    break;
+                }
+                description.append("<br>");
+                description.append(timeLine);
+            }
+            Matcher tf = TIME_FILE_PATTERN.matcher(timeLine);
+            if (tf.matches()) {
+                builder.setModuleName(tf.group(2));
+                builder.setFileName(tf.group(3));
+                builder.setLineStart(tf.group(4));
+            }
+            Matcher vsim = VSIM_PATTERN.matcher(message);
+            if (vsim.matches()) {
+                builder.setCategory(vsim.group(1));
+                builder.setFileName(vsim.group(2));
+                builder.setLineStart(vsim.group(3));
+                message = vsim.group(4);
+            }
+        }
+
+        builder.setMessage(message);
+        builder.setDescription(description.toString());
+        builder.setOrigin("Modelsim");
```

Comment:
`setOrigin` should not be used (as it will be overwritten in the warnings-ng plugin).